taxi/views.py

```diff
@@ -1,47 +1,54 @@
 from django.shortcuts import render
 from django.views import generic
+from django.contrib.sessions.models import Session
+from django.contrib.auth.decorators import login_required  # for functions
+from django.contrib.auth.mixins import LoginRequiredMixin  # for classes
 
+from django.http import HttpRequest, HttpResponse
 from .models import Driver, Car, Manufacturer
 
 
-def index(request):
+@login_required
+def index(request: HttpRequest) -> HttpResponse:
     """View function for the home page of the site."""
 
     num_drivers = Driver.objects.count()
     num_cars = Car.objects.count()
     num_manufacturers = Manufacturer.objects.count()
-
+    num_visits = request.session.get("num_visits", 0) + 1
```

Comment:
Consider using the `pluralize` template filter in your template to handle singular and plural forms of 'time' correctly. This will make your code more efficient and align with the checklist requirement.